provider/pkg/resources/cfn_custom_resource.go

```diff
@@ -241,23 +241,20 @@ func (c *cfnCustomResource) Check(ctx context.Context, urn urn.URN, randomSeed [
 
 	var failures []ValidationFailure
 
-	if !lambdaFunctionArnRegex.MatchString(typedInputs.ServiceToken) {
+	serviceTokenInput, hasServiceToken := inputs[resource.PropertyKey("serviceToken")]
+	if (!hasServiceToken || !serviceTokenInput.ContainsUnknowns()) && !lambdaFunctionArnRegex.MatchString(typedInputs.ServiceToken) {
 		failures = append(failures, ValidationFailure{
 			Path:   "serviceToken",
 			Reason: "serviceToken must be a valid Lambda function ARN.",
 		})
 	}
 
-	if typedInputs.StackID == nil {
+	_, hasStackID := inputs[resource.PropertyKey("stackId")]
+	if !hasStackID && typedInputs.StackID == nil {
 		// if the stack ID is not provided, we use the pulumi stack ID as the stack ID
 		inputs[resource.PropertyKey("stackId")] = resource.NewStringProperty(urn.Stack().String())
 	}
 
-	if typedInputs.CustomResourceProperties != nil {
```

Comment:
I decided to move `stringifying` the input `customResourceProperties` into the Create/Update/Delete methods.

This is an implementation detail and shouldn't really be done in `Check`. It would also complicate handling secrets and unknowns.